server/sonar-ce/src/test/java/org/sonar/ce/cluster/HazelcastClientWrapperTest.java

```diff
@@ -0,0 +1,137 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.ce.cluster;
+
+import com.hazelcast.core.HazelcastInstance;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.DisableOnDebug;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TestRule;
+import org.junit.rules.Timeout;
+import org.sonar.api.config.MapSettings;
+import org.sonar.api.config.PropertyDefinitions;
+import org.sonar.api.config.Settings;
+import org.sonar.process.NetworkUtils;
+import org.sonar.process.ProcessProperties;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.data.MapEntry.entry;
+
+public class HazelcastClientWrapperTest {
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  @Rule
+  public TestRule safeGuard = new DisableOnDebug(Timeout.seconds(10));
+
+  @Test
+  public void missing_CLUSTER_ENABLED_throw_ISE() {
+    expectedException.expect(IllegalStateException.class);
+    expectedException.expectMessage("Cluster is not enabled");
+
+    Settings settings = createClusterSettings("sonarqube", "localhost:9003");
+    settings.removeProperty(ProcessProperties.CLUSTER_ENABLED);
+    HazelcastClientWrapper hazelcastClientWrapper = new HazelcastClientWrapper(settings);
+    hazelcastClientWrapper.stop();
+  }
+
+  @Test
+  public void missing_CLUSTER_NAME_throw_ISE() {
+    expectedException.expect(IllegalStateException.class);
+    expectedException.expectMessage("sonar.cluster.name is missing");
+
+    Settings settings = createClusterSettings("sonarqube", "localhost:9003");
+    settings.removeProperty(ProcessProperties.CLUSTER_NAME);
+    HazelcastClientWrapper hazelcastClientWrapper = new HazelcastClientWrapper(settings);
+    hazelcastClientWrapper.stop();
+  }
+
+  @Test
+  public void missing_CLUSTER_LOCALENDPOINT_throw_ISE() {
+    expectedException.expect(IllegalStateException.class);
+    expectedException.expectMessage("LocalEndPoint have not been set");
+
+    Settings settings = createClusterSettings("sonarqube", "localhost:9003");
+    settings.removeProperty(ProcessProperties.CLUSTER_LOCALENDPOINT);
+    HazelcastClientWrapper hazelcastClientWrapper = new HazelcastClientWrapper(settings);
+    hazelcastClientWrapper.stop();
+  }
+
+  @Test
+  public void client_must_connect_to_hazelcast() {
```

Comment:
I would name this method `start_starts_hazelcast_client_and_connectst_to_cluster_specified_in_properties`